sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeBatchActionsAsyncClient.java

```diff
@@ -0,0 +1,445 @@
+                })
+                .collect(Collectors.toList()));
+        }
+        final AnalyzeBatchActionsResult analyzeBatchActionsResult = new AnalyzeBatchActionsResult();
+
+        // TODO:
+        //  Partial complete is still not well functional, https://github.com/Azure/azure-sdk-for-java/issues/18897
+        //  Error index should map back to input tasks order. Currently, the target has reference and the
+        //  document result is object without error and value
+//        final List<TextAnalyticsError> errors = analyzeJobState.getErrors();
+//        if (!CoreUtils.isNullOrEmpty(errors)) {
+//            final TextAnalyticsException textAnalyticsException = new TextAnalyticsException(
+//                "Analyze operation failed", null, null);
+//            final IterableStream<com.azure.ai.textanalytics.models.TextAnalyticsError> textAnalyticsErrors =
+//                IterableStream.of(errors.stream().map(Utility::toTextAnalyticsError).collect(Collectors.toList()));
+//            TextAnalyticsExceptionPropertiesHelper.setErrors(textAnalyticsException, textAnalyticsErrors);
+//            throw logger.logExceptionAsError(textAnalyticsException);
+//        }
+
+        final RequestStatistics requestStatistics = analyzeJobState.getStatistics();
+        TextDocumentBatchStatistics batchStatistics = null;
+        if (requestStatistics != null) {
+            batchStatistics = new TextDocumentBatchStatistics(
+                requestStatistics.getDocumentsCount(), requestStatistics.getErroneousDocumentsCount(),
+                requestStatistics.getValidDocumentsCount(), requestStatistics.getTransactionsCount()
+            );
+        }
+
+        AnalyzeBatchActionsResultPropertiesHelper.setStatistics(analyzeBatchActionsResult, batchStatistics);
+        AnalyzeBatchActionsResultPropertiesHelper.setRecognizeEntitiesActionResults(analyzeBatchActionsResult,
+            recognizeEntitiesActionResults);
+        AnalyzeBatchActionsResultPropertiesHelper.setRecognizePiiEntitiesActionResults(analyzeBatchActionsResult,
+            recognizePiiEntitiesActionResults);
+        AnalyzeBatchActionsResultPropertiesHelper.setExtractKeyPhrasesActionResults(analyzeBatchActionsResult,
+            extractKeyPhrasesActionResults);
+        return analyzeBatchActionsResult;
+    }
+
+    private Mono<PollResponse<AnalyzeBatchActionsOperationDetail>> processAnalyzedModelResponse(
+        Response<AnalyzeJobState> analyzeJobStateResponse,
+        PollResponse<AnalyzeBatchActionsOperationDetail> operationResultPollResponse) {
+
+        LongRunningOperationStatus status = LongRunningOperationStatus.SUCCESSFULLY_COMPLETED;
+        if (analyzeJobStateResponse.getValue() != null && analyzeJobStateResponse.getValue().getStatus() != null) {
+            switch (analyzeJobStateResponse.getValue().getStatus()) {
+                case CANCELLING:
+                    status = LongRunningOperationStatus.fromString("CANCELLING", false);
```

Comment:
We should use the string value from the status instead of the string "CANCELLING" that way we don't have to maintain the value locally and use whatever the service is passing.

Reply:
Same for some others below.